dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,569 @@
+                                        : userAPI.loadUserById(decryptedId, systemUser, false);
+
+
+                        String userIdFromCAS = (String) request.getSession(false)
+                                        .getAttribute("edu.yale.its.tp.cas.client.filter.user");
+
+                        if (UtilMethods.isSet(userIdFromCAS)) {
+                            CASAuthUtils.setUserValuesOnSession(user, request, response, true);
+                        }
+
+                        return true;
+
+                    } catch (Exception ex) {
+                        return false;
+                    }
+                } else
+                    Logger.info(this.getClass(), "Unable to retrieve user from LDAP/CAS with id: "
+                                    + decryptedId);
+            }
+
+            doLogout(request, response);
+
+            return false;
+
+        }
+    }
+
+
+
+    @Override
+    public boolean doCookieLogin(final String encryptedId, final HttpServletRequest request,
+                    final HttpServletResponse response) {
+        // note: keep in mind we are doing BE and FE login, not sure if this is right
+        final boolean doCookieLogin = _doCookieLogin(encryptedId, request, response);
+
+
+
+        return doCookieLogin;
+    }
+
+    @Override
+    public boolean doLogin(final String userName, final String password, final boolean rememberMe,
+                    final HttpServletRequest request, final HttpServletResponse response)
+                    throws NoSuchUserException {
+
+        return _doLogin(userName, password, rememberMe, request, response, false);
+    }
+
+
+
+    private boolean _doLogin(final String userIdOrEmail, final String password,
+                    final boolean rememberMe, final HttpServletRequest request,
+                    final HttpServletResponse response, final boolean skipPasswordCheck)
+                    throws NoSuchUserException {
+
+        LoginEvent event = new LoginEvent(this, LoginEvents.PRE_LOGIN, userIdOrEmail, password,
+                        request, response);
+        for (LoginEventListener listener : _listeners) {
+            listener.loginEventReceived(event);
+        }
+
+        boolean auth = _doLoginInternal(userIdOrEmail, password, rememberMe, request, response,
+                        false);
+
+        event = (auth) ? new LoginEvent(this, LoginEvents.POST_LOGIN_SUCCESS, userIdOrEmail,
+                        password, request, response)
+                        : new LoginEvent(this, LoginEvents.POST_LOGIN_FAILURE, userIdOrEmail,
+                                        password, request, response);
+        for (LoginEventListener listener : _listeners) {
+            listener.loginEventReceived(event);
+        }
+
+        return auth;
+
+
+    }
+
+
+
+    private boolean _doLoginInternal(final String userIdOrEmail, final String password,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)